Source/WebKit/Shared/LogStream.cpp

```diff
@@ -43,12 +43,15 @@ namespace WebKit {
 
 LogStream::~LogStream()
 {
-    if (RefPtr logStreamConnection = m_logStreamConnection) {
-        logStreamConnection->stopReceivingMessages(Messages::LogStream::messageReceiverName(), m_logStreamIdentifier->toUInt64());
-        logStreamConnection->invalidate();
-    }
-    if (RefPtr logWorkQueue = m_logWorkQueue)
+    if (RefPtr logWorkQueue = m_logWorkQueue) {
```

Comment:
This cannot be like this.
LogStream is message receiver, receiving messages in other thread.
It cannot be that the other thread is still using this, while this is also in destructor. Makes no sense.
This has to be its own function.
See the users  of ScopedActiveMessageReceiveQueue, you need to hold LogStream with it.